engine/consensus/dkg/reactor_engine.go

```diff
@@ -201,6 +201,13 @@ func (e *ReactorEngine) startDKGForEpoch(currentEpochCounter uint64, first *flow
 		}
 	})
 
+	// flag that we are starting the dkg for this epoch
+	err = e.dkgState.SetDKGStarted(nextEpochCounter)
```

Comment:
I think we should still set `DKGStarted` before starting the DKG process. The purpose of this flag currently is to avoid restarting a DKG which we have already started, since we are unable to recover an in-progress DKG state. We can be sure of this if we set the flag before starting the DKG. 

Though it's unlikely that the DKG controller would eg. send a message and the node restart before setting the flag, it's less easy to reason about if we asynchronously start the DKG before setting the flag.

Reply:
https://github.com/onflow/flow-go/pull/1849/commits/ae869441320976a92be677daf76299ea178572c5